tools/rules-check/rules-check.cc

```diff
@@ -41,6 +41,7 @@ int main(int argc, char **argv) {
 
     if (*args == NULL) {
         print_help(argv[0]);
+        delete rules;
```

Comment:
How about making rules a std::unique_ptr instead which takes care of the lifetime

Reply:
Yep, seems that works (see next commit).